### PR TITLE
refractor GET /api/articles (topic query)

### PR DESCRIPTION
**Pull request by Geno419**

### CODE REVIEW

File: db/model/model.js [hunk 1]
```diff
@@ -152,3 +146,13 @@ exports.getArticlesByTopic = (topic = "PassBYChecker") => {
       .then(({ rows }) => rows);
   }
```
Comment (alice-y-codes) [inline db/model/model.js:147]: your sql query needs to be refactored so that you still get comments as well as your topics if a user has a topic query. currently you are doing either or which is not what we want

**Review by alice-y-codes — COMMENTED** [review]
Still missing a test for Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper
